src/walletSdk/horizon/Account.ts

```diff
@@ -0,0 +1,57 @@
+import { Keypair, Transaction, FeeBumpTransaction } from "stellar-sdk";
+
+class AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    this.keypair = keypair;
+  }
+
+  get publicKey(): string {
+    return this.keypair.publicKey();
+  }
+
+  toString(): string {
+    return `${this.constructor.name}(address=${this.publicKey})`;
+  }
+}
+
+export class PublicKeypair extends AccountKeypair {
+  keypair: Keypair;
+  constructor(keypair: Keypair) {
+    super(keypair);
+  }
+
+  static fromString = (str: string) => {
+    const kp = Keypair.fromPublicKey(str);
+    return new PublicKeypair(kp);
+  };
+}
+
+export class SigningKeypair extends AccountKeypair {
```

Comment:
should it extend `PublicKeypair` instead? I'd assume having a secret gives you the power to have all `PublicKeypair` functions as well

Reply:
In Kotlin, we have them split. So all methods either accept `AccountKeyPair` (so either will work), or strictly `SigningKeyPair`. 
I think your proposal is more or less the same as what we have right now, but probably better to keep differences as small as possible.